mpi-proxy-split/doc/nimrod-build-tutorial.txt

```diff
@@ -0,0 +1,273 @@
+  * Java
+  * GCC
+  * Python 3
+  * Additional packages
+
+3. Building NIMROD
+
+NIMROD has the following repository structure:
+
+.
++-- nimall
+|   +-- bilder
+|   +-- bilderconf
+|   +-- configscripts
+|       +-- NERSC
+|   +-- fciowrappers
+|   +-- nimdevel
+|   +-- nimtests
+|   +-- superlu_dist7
+|   +-- xdraw
+|   +-- BUILD
+|   +-- H5VISIT
+|   +-- MAIL
+|   +-- NERSC
+|   +-- SVN
+|   +-- TESTING
+|   +-- externalrepos.sh
+|   +-- mknimall.sh
+
+The nimall directory contains NIMROD's source code (in lowercase), documentation
+for building or testing (capitalized), and scripts that help the installation
+process.
+
+### ON CORI ###
+
+Up to date configure and build scripts are available on Cori - the directory
+containing these scripts can be found in configscripts/NERSC. Make a copy of
+these scripts in another directory, depending on whether you require a serial
+or parallel build:
+  * cori.nersc.gov-nimdevel-{ser,par}-build.sh
+  * cori.nersc.gov-nimdevel-{ser,par}-config.sh
+
+In this tutorial, we'll use the parallel build of NIMROD.
+
+In cori.nersc.gov-nimdevel-par-config.sh, you'll have to change two environment
+variables:
+
+  * -DCMAKE_INSTALL_PREFIX:PATH=...    # set to directory to install NIMROD in
+  * -DSUPRA_SEARCH_PATH:PATH='...' ... # set directory OUTSIDE apostrophes to
+                                       # where NIMROD source files are
+
+Then run the configure script, followed by the build script. This will install
+NIMROD in the former directory.
+
+### ON CENTOS 7 AND CENTOS STREAM ###
+
+## FIRST BUILD ##
+
+The initial build on other operating systems is done with the mknimall.sh
+script. There are a few flags of note; you may view the rest by running the
+script with the -h flag:
+
+  * -b <build_dir> ... Build in <build_dir>. By default, this is wherever nimall
+                       is.
+  * -F ............... Force installation of packages that have local
+                         modifications. On CentOS 7, there seems to be an issue
+                         whereby the build process modifies the SuperLU_DIST
+                         source as part of the build process, and the script
+                         doesn't install the package. Consequently, the
+                         parallel version of NIMROD can't find it to link, and
+                         is built without it, causing subsequent run-time
+                         problems.
+  * -f ............... Install all packages.
+  * -i <install_dir> . Set comma delimited list of installation directories for
+                         code in subdirs. By default, this is $HOME/software.
+  * -j <n> ........... Maximum number of build processes summed over all builds.
+                         NIMROD will take about 40 minutes to build if n=1.
+  * -t ............... Run tests after building.
+
+Run the mknimall.sh with the necessary flags. I've found the following to work
+best:
+
+  $ ./mknimall.sh -F -j 8 -t
+
```

Comment:
Instead of the next two lines, let's write:
After `The script creates a new directory, builds, in <build_dir>.` , add something like: `See below for comments on cases where Nimrod failed to build on the first try, and we had to then modify portions and re-build.`